tests/ops/qutrit/test_qutrit_parametric_ops.py

```diff
@@ -0,0 +1,331 @@
+            qml.TRX.compute_matrix(np.pi, subspace=[0, 1]), expected, atol=tol, rtol=0
+        )
+        assert np.allclose(
+            qml.TRX(np.pi, wires=0, subspace=[0, 1]).matrix(), expected, atol=tol, rtol=0
+        )
+
+        expected = -1j * np.array([[1j, 0, 0], [0, 0, 1], [0, 1, 0]])
+        assert np.allclose(
+            qml.TRX.compute_matrix(np.pi, subspace=[1, 2]), expected, atol=tol, rtol=0
+        )
+        assert np.allclose(
+            qml.TRX(np.pi, wires=0, subspace=[1, 2]).matrix(), expected, atol=tol, rtol=0
+        )
+
+        expected = -1j * np.array([[0, 0, 1], [0, 1j, 0], [1, 0, 0]])
+        assert np.allclose(
+            qml.TRX.compute_matrix(np.pi, subspace=[0, 2]), expected, atol=tol, rtol=0
+        )
+        assert np.allclose(
+            qml.TRX(np.pi, wires=0, subspace=[0, 2]).matrix(), expected, atol=tol, rtol=0
+        )
+
+
+# TODO: Add tests for grad
+
+label_data = [
+    (qml.TRX(1.23456, wires=0), "TRX", "TRX\n(1.23)", "TRX\n(1)", "TRX⁻¹\n(1)"),
+]
+
+label_data_broadcasted = [
+    (qml.TRX(np.array([1.23, 4.56]), wires=0), "TRX", "TRX", "TRX", "TRX⁻¹"),
+]
+
+
+class TestLabel:
+    """Test the label method on parametric ops"""
+
+    @pytest.mark.parametrize("op, label1, label2, label3, label4", label_data)
+    def test_label_method(self, op, label1, label2, label3, label4):
+        """Test label method with plain scalers."""
+
+        assert op.label() == label1
+        assert op.label(decimals=2) == label2
+        assert op.label(decimals=0) == label3
+
+        op.inv()
+        assert op.label(decimals=0) == label4
+        op.inv()
+
+    @pytest.mark.parametrize("op, label1, label2, label3, label4", label_data_broadcasted)
+    def test_label_method_broadcasted(self, op, label1, label2, label3, label4):
+        """Test label method with plain scalers."""
+
+        assert op.label() == label1
+        assert op.label(decimals=2) == label2
+        assert op.label(decimals=0) == label3
+
+        op.inv()
+        assert op.label(decimals=0) == label4
+        op.inv()
+
+    @pytest.mark.tf
+    def test_label_tf(self):
+        """Test label methods work with tensorflow variables"""
+        import tensorflow as tf
+
+        op1 = qml.TRX(tf.Variable(0.123456), wires=0)
+        assert op1.label(decimals=2) == "TRX\n(0.12)"
+
+    @pytest.mark.torch
+    def test_label_torch(self):
+        """Test label methods work with torch tensors"""
+        import torch
+
+        op1 = qml.TRX(torch.tensor(1.23456), wires=0)
+        assert op1.label(decimals=2) == "TRX\n(1.23)"
+
+    @pytest.mark.jax
+    def test_label_jax(self):
+        """Test the label method works with jax"""
+        import jax
+
+        op1 = qml.TRX(jax.numpy.array(1.23456), wires=0)
+        assert op1.label(decimals=2) == "TRX\n(1.23)"
+
+    def test_string_parameter(self):
+        """Test labelling works if variable is a string instead of a float."""
+
+        op1 = qml.TRX("x", wires=0)
```

Comment:
Shouldn't this fail if a non-numerical value is passed?

Reply:
That isn't necessary. The case where this would be expected to fail is if `TRX` was being used in a device, or more specifically if the `TRX.compute_matrix` method was called, in which case the parameter cannot be used to compute a sine or cosine value and the `qml.math` module would throw an error. However, as far as labelling goes, this is expected behaviour. I've also confirmed by comparing `TRX` with `RX`.